webdriver/tests/bidi/network/add_intercept/invalid.py

```diff
@@ -0,0 +1,171 @@
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.add_intercept(
+            phases=["beforeRequestSent"],
+            url_patterns=[{"type": "string", "pattern": value}],
+        )
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        "foo",
+        "*",
+        "(",
+        ")",
+        "{",
+        "}",
+        "http\\{s\\}://example.com",
+        "https://example.com:port/",
+    ],
+)
+async def test_params_url_patterns_string_pattern_invalid_value(bidi_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.add_intercept(
+            phases=["beforeRequestSent"],
+            url_patterns=[{"type": "string", "pattern": value}],
+        )
+
+
+@pytest.mark.parametrize(
+    "property", ["protocol", "hostname", "port", "pathname", "search"]
+)
+@pytest.mark.parametrize("value", [False, 42, [], {}])
+async def test_params_url_patterns_pattern_property_invalid_type(
+    bidi_session, property, value
+):
+    with pytest.raises(error.InvalidArgumentException):
+        url_pattern = {"type": "pattern"}
+        url_pattern[property] = value
+        await bidi_session.network.add_intercept(
+            phases=["beforeRequestSent"],
+            url_patterns=[url_pattern],
+        )
+
+
+@pytest.mark.parametrize(
+    "property", ["protocol", "hostname", "port", "pathname", "search"]
+)
+@pytest.mark.parametrize("value", ["*", "(", ")", "{", "}"])
+async def test_params_url_patterns_pattern_property_unescaped_character(
+    bidi_session, property, value
+):
+    with pytest.raises(error.InvalidArgumentException):
+        url_pattern = {"type": "pattern"}
+        url_pattern[property] = value
+        await bidi_session.network.add_intercept(
+            phases=["beforeRequestSent"],
+            url_patterns=[url_pattern],
+        )
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        "http/",
+        "http\\*",
+        "http\\(",
+        "http\\)",
+        "http\\{",
+        "http\\}",
+        "http#",
+        "http@",
+        "http%",
+    ],
+)
+async def test_params_url_patterns_pattern_protocol_invalid_value(bidi_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.add_intercept(
+            phases=["beforeRequestSent"],
+            url_patterns=[{"type": "pattern", "protocol": value}],
+        )
+
+
+@pytest.mark.parametrize("value", ["abc/com/", "abc?com", "abc#com", "abc:com"])
+async def test_params_url_patterns_pattern_hostname_invalid_value(bidi_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.add_intercept(
+            phases=["beforeRequestSent"],
+            url_patterns=[{"type": "pattern", "hostname": value}],
+        )
+
+
+@pytest.mark.parametrize("value", ["abcd", "-1", "80 ", "1.3", ":80"])
```

Comment:
maybe add 65536 as well? (one value above the maximum)

Reply:
Good point! Added.